### PR TITLE
feat: add ui components (UiInput, UiDropdown and PointerEvents)

### PR DESCRIPTION
**Pull request by leanmendoza**
## Complete implementation of UI by adding:
- [x] `PointerEvents`
- [x] `UiDropdown`
- [x] `UiInput`

### CODE REVIEW

File: rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+#[godot_api]
+impl DclUiControl {
+    pub fn new_alloc() -> Gd<Self> {
+        Gd::new_default()
+    }
+
+    #[func]
+    pub fn _on_gui_input(&mut self, input: Gd<InputEvent>) {
+        let global_tick_number = unsafe { *GLOBAL_TICK_NUMBER.as_ptr() as u32 };
```
Comment (kuruk-mm) [inline rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs:68]:
```suggestion
        let global_tick_number = GLOBAL_TICK_NUMBER.load(Ordering::Relaxed);
```

File: rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                            timestamp: global_tick_number,
+                            analog: None,
+                            tick_number: global_tick_number,
+                        },
+                    ));
+                }
+            }
+        }
+
+        // TODO: it enables HOVER and LEAVE events
```
Comment (kuruk-mm) [inline rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs:104]: What is the status of this comment code?

File: rust/decentraland-godot-lib/src/scene_runner/scene_manager.rs [hunk 3]
```diff
@@ -69,12 +68,14 @@ pub struct SceneManager {
...
     input_state: InputState,
     last_raycast_result: Option<GodotDclRaycastResult>,
-    global_tick_number: u32,
 
     #[export]
     pointer_tooltips: VariantArray,
 }
 
+// This value is the current global tick number, is used for marking the cronolgy of lamport timestamp
+pub static GLOBAL_TICK_NUMBER: AtomicU64 = AtomicU64::new(0);
```
Comment (kuruk-mm) [inline rust/decentraland-godot-lib/src/scene_runner/scene_manager.rs:77]:
```suggestion
pub static GLOBAL_TICK_NUMBER: AtomicU32 = AtomicU32::new(0);
```

File: rust/xtask/src/main.rs [hunk 4]
```diff
@@ -100,6 +101,7 @@ fn main() -> Result<(), anyhow::Error> {
     let root = xtaskops::ops::root_dir();
     let res = match subcommand {
         ("install", sm) => install_dependency::install(sm.is_present("no-templates")),
+        ("update-protocol", _) => install_dependency::install_dcl_protocol(),
```
Comment (kuruk-mm) [inline rust/xtask/src/main.rs:104]: Looks like the `update-protocol.rs` is missing :thinking:

**Review by kuruk-mm — APPROVED** [review]
Looks great, some comments to fix before merging

File: rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs (hunk 2 above)
Comment (leanmendoza) [reply rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs:104]: It isn't specified in the ADR and not implemented in the SDK, if we add it, no one could use it

**Pull request merged**